Use .strip_heredoc to remove indents in heredocs

edwardloveall opened the pull request:
Useful for embedding code snippets in tests or view helpers (and other places I'm sure).

edwardloveall committed:
Use .strip_heredoc to remove indents in heredocs

Useful for embedding code snippets in tests or view helpers (and other places I'm sure).

bf4 commented on rails/strip_heredoc.md:
````diff
@@ -0,0 +1,52 @@
+If you need to load this outside of rails:
+
+```ruby
+require 'active_support/core_ext/string/strip'
````
or just e.g. https://github.com/mikel/mail/blob/0f9393bb3ef1344aa76d6dac28db3a4934c65087/spec/spec_helper.rb#L57-L61

``` ruby
# https://github.com/rails/rails/blob/master/activesupport/lib/active_support/core_ext/string/strip.rb#L22
def strip_heredoc(string)
  indent = string.scan(/^[ \t]*(?=\S)/).min.size
  string.gsub(/^[ \t]{#{indent}}/, '')
end unless String.method_defined?(:strip_heredoc)
```

edwardloveall replied:
I put this in the rails category because I first encountered this in test for a rails app, and it's also defined by ActiveSupport. I've always seen this repo as _What we can do with what already exists_ instead of _Solve this particular programming problem with new code_.

What do you think of that?

bf4 replied:
Hey, no worries, my intent was just to append info to `If you need to load this outside of rails:`

edwardloveall replied:
Yeah good point. I'll try and rephrase or link to that code or something.

edwardloveall replied:
Added below.

edwardloveall committed:
Include link to strip_heredoc source

Since the method is very small and self contained, it might make sense for the developer to just copy/paste it into the project. I added it as a link so it was less emphasized. I feel like it's better to just use what already exists, but sometimes that doesn't make sense.

edwardloveall commented:
Honestly, a better way to do this now seems to use the new [indent heredoc](https://bugs.ruby-lang.org/issues/9098#note-6): `<<~`

But this repo seems to be abandoned. If wrong about that, someone let me know and I'll make a new PR.

stefannibrasil commented:
This repo is no longer active and it's being archived. The contents will still be available as read-only. Thank you for your contributions!
